### ISSUE TITLE
Provider public key well-formedness proof still missing

### ISSUE THREAD

**Issue by pschuermann97** [cryptimeleon/incentive-system#34]
We omitted the ZKP of the well-formedness of the provider public key in the beginning of the Issue-Join protocol for now.
We should add it later (where to do so will be discusseed on May 21, 2021)

**Comment by this-kramer** [comment]
@JanBobolz is this covered by using TLS and thus having an authenticated connection?

**Comment by JanBobolz** [comment]
No, this proof is to force the provider to generate keys honestly (in some sense), not to make sure that the key is being transmitted correctly. 

It should not become part of Issue/Join. Rather, I'd recommend adding a Fiat-Shamir proof to the provider public key.
